### ISSUE TITLE
Cannot download esim from Google Fi

### ISSUE THREAD

**Issue by opokadot** [GrapheneOS/os-issue-tracker#3191]
Cannot download esim from Google Fi. 
Google Fi downloads sim from its app not from Android settings. 

Tried clearing caches, reinstalling, factory reset etc etc

Google Fi & system logs

[Google Fi log 94d5f4f076b2.txt](https://github.com/GrapheneOS/os-issue-tracker/files/14184951/Google.Fi.log.94d5f4f076b2.txt)
[System log 3b94e27405c1.txt](https://github.com/GrapheneOS/os-issue-tracker/files/14184952/System.log.3b94e27405c1.txt)

**Comment by muhomorr** [comment]
Does Google Fi app have the Phone permission?

**Comment by opokadot** [comment]
> Does Google Fi app have the Phone permission?

Yes all permissions enabled.

**Comment by muhomorr** [comment]
Google Fi log says that it doesn't have the Phone permission. Did Google Fi have Phone permission at the time log was captured? Does Play services app have the Phone permission?

**Comment by opokadot** [comment]
> Google Fi log says that it doesn't have the Phone permission. Did Google Fi have Phone permission at the time log was captured? Does Play services app have the Phone permission?

Yes it was given every permission and allowed all with no scopes. Google services also and play and Google app, everything google

**Comment by opokadot** [comment]
> Google Fi log says that it doesn't have the Phone permission. Did Google Fi have Phone permission at the time log was captured? Does Play services app have the Phone permission?

I was thinking the sandbox was causing it somehow

**Comment by WinstonHartnett** [comment]
I'm also experiencing this issue. I've given Google Play Services, Google Play, and Google Fi all permissions. Same `READ_PHONE_STATE permission is missing` shows up in log. eSIM support is enabled.

**Comment by WinstonHartnett** [comment]
> > I'm also experiencing this issue. I've given Google Play Services, Google Play, and Google Fi all permissions. Same READ_PHONE_STATE permission is missing shows up in log. eSIM support is enabled.
> 
> READ_PHONE_STATE is the Phone permission. Seems you haven't granted it to an app needing it.

In Settings > Apps > Google Fi > Permissions, I've enabled:
- Call logs
- Contacts
- Location
- Microphone
- Network
- Notifications
- Phone (which was "last accessed" when I opened the app)
- Sensors
- SMS

This, as far as I know, is every permission in that menu. I've similarly enabled every listed permission for Google Play and Play Services.

**Comment by WinstonHartnett** [comment]
eSIM is definitely broken. Just did a factory reset and tried again --- doesn't work.

**Comment by WinstonHartnett** [comment]
At the same menu where Google Fi fails on GOS, the stock [Pixel 8] OS displays a pop-up asking for carrier permissions. There wasn't any pop-up on GOS (Google Fi immediately fails). The logs linked above have this line that may be related:

```
         1707245516.021 19374 19944 I Tycho.jnk: First call to TelephonyManager.hasCarrierPrivileges in process returned false
```

**Comment by opokadot** [comment]
I'm still having this issue as well, glad this was brought up by another person (not glad someone else has to deal with this problem lol)

**Comment by billyshits** [comment]
Definitely doesn’t work. Tried different reflashing, two different pixels (6a and fold), and made sure all permissions were on. Even downloaded Visible to see if I could set up an eSIM from that carrier. Nope. Something is wrong with eSIM and it’s not just Fi.

**Comment by opokadot** [comment]
No update on this? It can't be a terribly difficult fix

**Comment by opokadot** [comment]
No

**Comment by marsposting** [comment]
I'm still running into this on a Pixel 8 Pro with the current beta  (2024042200), privileged eSIM turned on, and all permissions enabled for the Google Fi app. The error code is T120, which I believe means that the app can't access the eSIM manager based on other comments elsewhere.

**Comment by spikebike** [comment]
I have a pixel 6 and pixel 7 with GrapheneOS + Google FI working great, they were installed around a year ago.   I tried to install the current (installed yesterday) and can't get Google Fi working.  I tried various SIM+Esim variations, rebooting, and verifying that *#*#4636#*#* -> DSDS enabled.  Redownloading the dim, using a QR code for Google FI to load an eSim, disabling/re-enabling eSim, giving all permissions, etc didn't work.

**Comment by OneDeuxTriSeiGo** [comment]
I've been tracking this issue as I've had the same problem as everyone else here. I'm on the Pixel 6 pro fwiw. I just fetched the alpha release.

It went through the eSIM provisioning okay. Calls, SMS, RCS, and data all seem to be working okay.

**Comment by spikebike** [comment]
I joined the beta channel yesterday, nothing happened that I could tell.  This morning (May 4th, 2024) I got notified of a new version of Graphene, I upgraded and rebooted.  Google Fi didn't work, clicked on the registration with Google Fi, didn't work.

I deleted the Google Fi SIM, verified eSIM was enabled, and restarted google Fi.  That worked.  I got fast connectivity, SMS messages, and Google Fi seems happy.

Many thanks, my phone is back to 100%.

**Comment by alexsapps** [comment]
Google Fi support helped me fix error T120 while activating Google Fi on GrapheneOS with eSIM. Here's what I did:

* Google Fi app settings -> Mobile data usage -> turn on "Unrestricted data usage"
* Google Fi app settings -> Permissions -> turn on Phone, SMS, and Location
* Phone settings -> Network & internet -> turn on "eSIM support"

I had also turned on a bunch of permissions for Google Play Services and Google Fi app before contacting support. Not sure if those were necessary.
